packages/flutter_tools/lib/src/commands/precache.dart

```diff
@@ -55,7 +57,7 @@ class PrecacheCommand extends FlutterCommand {
         requiredArtifacts.add(artifact);
       }
     }
-    if (cache.isUpToDate()) {
+    if (cache.isUpToDate() && !argResults['force']) {
```

Comment:
nit: maybe easier to read flipped around:
```
final bool forceUpdate = argResult['force'];
if (forceUpdate || !cache.isUpToDate()) {
  await cache.updateAll(requiredArtifacts);
} else {
  printStatus(...);
}
```
Then you also don't evaluate cache.isUpToDate() when you have --force

Reply:
Perfect!